pkg/streamingpromql/engine_test.go

```diff
@@ -88,6 +87,15 @@ func TestUnsupportedPromQLFeaturesWithFeatureToggles(t *testing.T) {
 
 		requireRangeQueryIsUnsupported(t, featureToggles, "metric{} + other_metric{}", "binary expressions")
 		requireInstantQueryIsUnsupported(t, featureToggles, "metric{} + other_metric{}", "binary expressions")
+
+		requireRangeQueryIsUnsupported(t, featureToggles, "metric{} + 1", "binary expressions")
```

Comment:
(nit) etc..
```suggestion
		requireRangeQueryIsUnsupported(t, featureToggles, "metric{} + 1", "binary expressions with scalar and instant")
```

Reply:
Adding this would complicate [the code](https://github.com/grafana/mimir/blob/6a54260008a00d5c848eb0870527a2d897a0d09f/pkg/streamingpromql/query.go#L182) that generates the error message, and given this message will now only be generated when the feature toggle is turned off, I'm not sure it's worth it.